pkg/network/protocols/http2/protocol.go

```diff
@@ -353,45 +361,24 @@ func (p *protocol) GetStats() *protocols.ProtocolStats {
 	}
 }
 
-// UpdateKernelTelemetry should be moved to the HTTP/2 part as well
-func (p *protocol) UpdateKernelTelemetry(mgr *manager.Manager) {
+// updateKernelTelemetry updates the HTTP/2 kernel telemetry.
+func (p *protocol) updateKernelTelemetry(mgr *manager.Manager) {
 	var zero uint32
+	http2Telemetry := &HTTP2Telemetry{}
```

Comment:
once again, you're reallocating and getting the map in every iteration

Reply:
related old comments -
* https://github.com/DataDog/datadog-agent/pull/21035#discussion_r1402886469
* https://github.com/DataDog/datadog-agent/pull/21035#discussion_r1402886384

Reply:
changed